app/views/layouts/application.html.haml

```diff
@@ -1,12 +1,28 @@
-%html
+!!!
+%html{lang: "en"}
   %head
-    %title Todo
-    =stylesheet_link_tag 'application', media: 'all', 'data-turbolinks-track' => true
-    =javascript_include_tag 'application', 'data-turbolinks-track' => true
-    =csrf_meta_tags
+    = stylesheet_link_tag 'application', media: 'all', 'data-turbolinks-track' => true
+    = javascript_include_tag 'application', 'data-turbolinks-track' => true
+    = csrf_meta_tags
+    %meta{charset: "utf-8"}
+    %meta{content: "width=device-width, initial-scale=1", name: "viewport"}
+    %link{crossorigin: "anonymous", href: "https://cdn.jsdelivr.net/npm/bootstrap@5.0.0-beta1/dist/css/bootstrap.min.css", integrity: "sha384-giJF6kkoqNQ00vy+HMDP7azOuL0xtbfIcaT9wjKHr8RbDVddVHyTfAAsrekwKmP1", rel: "stylesheet"}
+    %title TODO
   %body
+    - if current_user
+      = render 'layouts/navbar'
     #flash-display
-    - flash.each do |type, msg|
-      %div
-        = msg
-    =yield
+      - flash.each do |type, msg|
+        - if type == 'notice'
+          %div{class: "alert alert-success"}
+            =msg
+            %button.btn-close{"aria-label" => "Close", type: "button"}
+        - else
+          %div{class: "alert alert-#{type}"}
+            = msg
+            %button.btn-close{"aria-label" => "Close", type: "button"}
+
+
+    .container=yield
+
+    %script{crossorigin: "anonymous", integrity: "sha384-ygbV9kiqUc6oa4msXn9868pTtWMgiQaeYH7/t7LECLbyPA2x65Kgf80OJFdroafW", src: "https://cdn.jsdelivr.net/npm/bootstrap@5.0.0-beta1/dist/js/bootstrap.bundle.min.js"}
```

Comment:
When dealing with public scripts like these, do you have a preference between storing a vendored version locally vs fetching straight from the source?

Reply:
You could get the bootstrap-sass gem or whatever. I don't like taking this approach as it adds another layer to the gem dependency graph and the nightmare which ensues. Ideally, you could improve the above approach so that you could configure what scripts it's pulling in from. I just didn't want to bloat the code base just to get some basic bootstrap. Ideally, you'd be sourcing bootstrap from CDN so that the rails app doesn't have to handle it and so that rake assets:precompile takes less time. You'd only be compiling the changes you make to bootstrap. It also requires you having a connection to the internet if you want to test changes and it can't get to the cdn file.